x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/CacheService.java

```diff
@@ -459,21 +474,28 @@ protected void synchronizeCache() {
                         final Path cacheDir = cacheFilePath.toAbsolutePath().getParent();
                         if (cacheDirs.add(cacheDir)) {
                             try {
-                                IOUtils.fsync(cacheDir, true, false);
+                                IOUtils.fsync(cacheDir, true, false); // TODO evict cache file if fsync failed
```

Comment:
In relation to the todo, could we instead avoid persisting those that fail into the `PersistentCache`? We should be able to check `cacheDirs.contains(cacheDir)` before fsync'ing the dir and only add to `cacheDirs` if fsync is successful?

Reply:
Makes sense, I pushed eab0a0caf48c8996e3f17438e5621a5bac2c9ccd.